packages/query-core/src/mutation.ts

```diff
@@ -171,8 +173,8 @@ export class Mutation<
           }
           return this.options.mutationFn(this.state.variables!)
         },
-        onFail: () => {
-          this.dispatch({ type: 'failed' })
+        onFail: (_failureCount, error) => {
+          this.dispatch({ type: 'failed', error })
```

Comment:
I thought we'd also wanted to dispatch the failureCount here as well and _not_ increment it in the reducer?

```
this.dispatch({ type: 'failed', failureCount, error })
```

Reply:
This is what I meant with:
> failureCount implementation differs as it doesn't reset to 0 after a successful mutation, this led to not using the failureCount from the action but incrementing the one in the state

I didn't want to alter the behavior here. But if it is considered a bug I can fix it.

Reply:
I'm pretty sure this is not the intended behavior